rids/iocs/allowed_sni_port.py

```diff
@@ -12,31 +12,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
-Tests for the network_capture library.
-"""
+"""IOC parser for allowing certain unusual (server name, port) sightings."""
 
 
-import io
-import pytest
+from rids.iocs.iocs import RuleSet
+from rids.monitors.tls_monitor import TlsRule
 
-from rids import network_capture
 
+class AllowedEndpoints:
+  """Parses allowed (sni, port) pairs directly from an ioc_sources config."""
+  def __init__(self, config):
+    self.name = config['name']
+    self.allowlist = config['allow']
 
-@pytest.fixture()
-def badips_map():
-  return {
-    '1.1.1.1': 'Source A',
-    '4.5.6.7': 'Best IOCs',
-    '100.12.34.56': 'Threatbusters',
-  }
-
-
-def test_good_ip_is_ok(badips_map):
-  detected = network_capture.detect_bad_ips('0\t127.0.0.1\t1.2.3.4', badips_map)
-  assert not detected
-
-
-def test_bad_ip_is_logged(badips_map):
-  detected = network_capture.detect_bad_ips('1\t4.5.6.7\t127.0.0.1', badips_map)
-  assert detected
+  def provide_rules(self, ruleset: RuleSet):
+    for sni, port in self.allowlist:
+      tls_rule = TlsRule(
+          allowed_sni=sni,
+          expected_port=port)
+      ruleset.add_tls_rule(tls_rule)
```

Comment:
Up to you: You may be able to get rid of the TlsRule object and just call `tls_matcher.add_allowed_endpoint(sni, port)`

Reply:
done